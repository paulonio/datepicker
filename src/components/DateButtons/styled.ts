import styled, { css } from 'styled-components';

export const Week = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-wrap: wrap;
  color: #333333;
  font-size: 13px;
  font-weight: 600;
  line-height: 18px;
`;

interface WeekDayProps {
  mode: 'selectedFrom' | 'selectedTo' | 'inRange' | '';
  isCurrentMonth: boolean;
  isToday: boolean;
}

export const WeekDay = styled.button<WeekDayProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  background-color: #ffffff;
  cursor: pointer;
  color: ${({ isCurrentMonth }) => !isCurrentMonth && '#aaaaaa'};
  transition: background-color 0.3s, border-radius 0.3s, color 0.3s;
  &:hover {
    background-color: #f1f1f1;
    border-radius: 8px;
    color: #333333;
  }
  ${(props) => {
    switch (props.mode) {
      case 'selectedFrom':
        return css`
          background-color: rgba(47, 128, 237, 0.6);
          color: #ffffff;
          border-radius: 8px 0px 0px 8px;
        `;
      case 'selectedTo':
        return css`
          background-color: #2f80ed;
          color: #ffffff;
          border-radius: 0px 8px 8px 0px;
        `;
      case 'inRange':
        return css`
          background-color: rgba(47, 128, 237, 0.1);
          color: #2f80ed;
        `;
      default:
        return css``;
    }
  }}
  color: ${({ isToday }) => isToday && 'red'};
`;
