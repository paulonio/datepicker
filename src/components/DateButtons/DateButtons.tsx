import React, { FC } from 'react';
import { Week, WeekDay } from './styled';
import { DisplayDate } from '../../utils/utils';

interface DateButtonsProps {
  dateFrom: Date | null;
  dateTo: Date | null;
  allDays: Array<DisplayDate>;
  onDateChange: (date: Date) => void;
  toggleCalendar: (bool: boolean) => void;
  minDate?: Date;
  maxDate?: Date;
  selectDate: (date: Date) => void;
  toggleModal: (value: boolean) => void;
}

export const getToday = (): Date => {
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  return today;
};

const isValidDate = (date: Date, minDate?: Date, maxDate?: Date) => {
  if (!minDate && !maxDate) {
    return false;
  }
  if (!minDate && maxDate && date >= maxDate) {
    return true;
  }
  if (!maxDate && minDate && date <= minDate) {
    return true;
  }
  if ((minDate && date <= minDate) || (maxDate && date >= maxDate)) {
    return true;
  }
  return false;
};

const DateButtons: FC<DateButtonsProps> = ({
  dateFrom,
  dateTo,
  allDays,
  onDateChange,
  toggleCalendar,
  minDate,
  maxDate,
  selectDate,
  toggleModal,
}) => {
  const today = getToday();

  const getMode = (date: Date) => {
    if (dateFrom && dateFrom.getTime() === date.getTime()) {
      return 'selectedFrom';
    }
    if (dateTo && dateTo.getTime() === date.getTime()) {
      return 'selectedTo';
    }
    if (dateFrom && dateTo && date > dateFrom && date < dateTo) {
      return 'inRange';
    }
    return '';
  };

  const handleButtonClick = (date: Date) => {
    onDateChange(date);
    toggleCalendar(false);
  };

  const handleToggleModal = (date: Date) => {
    selectDate(date);
    toggleModal(true);
  };

  return (
    <Week>
      {allDays.map(({ currentYear, currentMonth, currentDate, isCurrentMonth }) => {
        const date = new Date(currentYear, currentMonth, currentDate);
        const mode = getMode(date);
        const isDisabled = isValidDate(date, minDate, maxDate);
        return (
          <WeekDay
            key={`${currentYear}-${currentMonth}-${currentDate}`}
            // onClick={() => handleButtonClick(date)}
            onDoubleClick={() => handleToggleModal(date)}
            mode={mode}
            isCurrentMonth={isCurrentMonth}
            isToday={today.getTime() === date.getTime()}
            disabled={isDisabled}
          >
            {currentDate}
          </WeekDay>
        );
      })}
    </Week>
  );
};

export default DateButtons;
