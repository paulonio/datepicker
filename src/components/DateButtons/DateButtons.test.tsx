import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import type { DatepickerProps } from '@components/Datepicker/Datepicker';
import DateButtons from '@components/DateButtons/DateButtons';
import type { Init } from '@/types/types';

describe('DateButtons component', () => {
  const config: DatepickerProps = {
    start: 'mo',
    view: 'monthly',
    weekend: 'show',
    minDate: new Date(2022, 4, 11),
    maxDate: new Date(2024, 4, 11),
  };

  const state: Init = {
    currentCalendar: 'From',
    showCalendar: true,
    from: null,
    to: null,
  };

  const initialDate = new Date(2023, 4, 11);

  it('should render DateButtons component', () => {
    const { container } = render(
      <DateButtons
        state={state}
        config={config}
        newDate={initialDate}
        selectOneDate={false}
        selectedDate={initialDate}
        selectDate={() => {}}
        setSelectOneDate={() => {}}
        dispatch={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should correctly highlight date from and date to', () => {
    const from = new Date(2023, 4, 8);
    const to = new Date(2023, 4, 12);
    const newState = { ...state, from, to };

    const { container } = render(
      <DateButtons
        state={newState}
        config={config}
        newDate={initialDate}
        selectOneDate={false}
        selectedDate={initialDate}
        selectDate={() => {}}
        setSelectOneDate={() => {}}
        dispatch={() => {}}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
