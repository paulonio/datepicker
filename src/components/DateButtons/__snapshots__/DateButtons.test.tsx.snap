// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateButtons component should correctly highlight date from and date to 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  color: #333333;
  font-size: 13px;
  font-weight: 600;
  line-height: 18px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
}

.c1:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c1:disabled {
  cursor: default;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  border: solid 1px blue;
}

.c2:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c2:disabled {
  cursor: default;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  background-color: rgba(47,128,237,0.6);
  color: #ffffff;
  border-radius: 8px 0px 0px 8px;
}

.c3:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c3:disabled {
  cursor: default;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  background-color: rgba(47,128,237,0.1);
  color: #2f80ed;
}

.c4:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c4:disabled {
  cursor: default;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  background-color: rgba(47,128,237,0.1);
  color: #2f80ed;
  color: red;
}

.c5:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c5:disabled {
  cursor: default;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  background-color: #2f80ed;
  color: #ffffff;
  border-radius: 0px 8px 8px 0px;
}

.c6:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c6:disabled {
  cursor: default;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  color: #aaaaaa;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
}

.c7:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c7:disabled {
  cursor: default;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  color: #aaaaaa;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  border: solid 1px blue;
}

.c8:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c8:disabled {
  cursor: default;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      data-testid="date-button"
    >
      1
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      2
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      3
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      4
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      5
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      6
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      7
    </button>
    <button
      class="c3"
      data-testid="date-button"
    >
      8
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      9
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      10
    </button>
    <button
      class="c5"
      data-testid="date-button"
    >
      11
    </button>
    <button
      class="c6"
      data-testid="date-button"
    >
      12
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      13
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      14
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      15
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      16
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      17
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      18
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      19
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      20
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      21
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      22
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      23
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      24
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      25
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      26
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      27
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      28
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      29
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      30
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      31
    </button>
    <button
      class="c7"
      data-testid="date-button"
    >
      1
    </button>
    <button
      class="c7"
      data-testid="date-button"
    >
      2
    </button>
    <button
      class="c8"
      data-testid="date-button"
    >
      3
    </button>
    <button
      class="c8"
      data-testid="date-button"
    >
      4
    </button>
    <button
      class="c7"
      data-testid="date-button"
    >
      5
    </button>
    <button
      class="c7"
      data-testid="date-button"
    >
      6
    </button>
    <button
      class="c7"
      data-testid="date-button"
    >
      7
    </button>
    <button
      class="c7"
      data-testid="date-button"
    >
      8
    </button>
    <button
      class="c7"
      data-testid="date-button"
    >
      9
    </button>
    <button
      class="c8"
      data-testid="date-button"
    >
      10
    </button>
    <button
      class="c8"
      data-testid="date-button"
    >
      11
    </button>
  </div>
</div>
`;

exports[`DateButtons component should render DateButtons component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  color: #333333;
  font-size: 13px;
  font-weight: 600;
  line-height: 18px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
}

.c1:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c1:disabled {
  cursor: default;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  border: solid 1px blue;
}

.c2:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c2:disabled {
  cursor: default;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  color: red;
}

.c3:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c3:disabled {
  cursor: default;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  color: #aaaaaa;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
}

.c4:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c4:disabled {
  cursor: default;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 32.85px;
  height: 32px;
  border: none;
  border-radius: 4px;
  background-color: #ffffff;
  cursor: pointer;
  color: #aaaaaa;
  -webkit-transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  transition: background-color 0.3s,border-radius 0.3s,color 0.3s;
  border: solid 1px blue;
}

.c5:enabled:hover {
  background-color: #f1f1f1;
  border-radius: 8px;
  color: #333333;
}

.c5:disabled {
  cursor: default;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      data-testid="date-button"
    >
      1
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      2
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      3
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      4
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      5
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      6
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      7
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      8
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      9
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      10
    </button>
    <button
      class="c3"
      data-testid="date-button"
    >
      11
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      12
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      13
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      14
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      15
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      16
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      17
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      18
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      19
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      20
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      21
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      22
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      23
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      24
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      25
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      26
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      27
    </button>
    <button
      class="c2"
      data-testid="date-button"
    >
      28
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      29
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      30
    </button>
    <button
      class="c1"
      data-testid="date-button"
    >
      31
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      1
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      2
    </button>
    <button
      class="c5"
      data-testid="date-button"
    >
      3
    </button>
    <button
      class="c5"
      data-testid="date-button"
    >
      4
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      5
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      6
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      7
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      8
    </button>
    <button
      class="c4"
      data-testid="date-button"
    >
      9
    </button>
    <button
      class="c5"
      data-testid="date-button"
    >
      10
    </button>
    <button
      class="c5"
      data-testid="date-button"
    >
      11
    </button>
  </div>
</div>
`;
